jwql/instrument_monitors/nircam_monitors/claw_monitor.py

```diff
@@ -0,0 +1,394 @@
+        self.query_table = eval('NIRCamClawQueryHistory')
+        self.stats_table = eval('NIRCamClawStats')
+
+    def make_background_plots(self):
+        """Makes plots of the background levels over time in NIRCam data.
+        """
+
+        # Get all of the background data.
+        query = session.query(NIRCamClawStats.filename, NIRCamClawStats.filter, NIRCamClawStats.pupil, NIRCamClawStats.detector,
+                              NIRCamClawStats.effexptm, NIRCamClawStats.expstart_mjd, NIRCamClawStats.entry_date, NIRCamClawStats.mean,
+                              NIRCamClawStats.median, NIRCamClawStats.frac_masked).all()
+        df_orig = pd.DataFrame(query, columns=['filename', 'filter', 'pupil', 'detector', 'effexptm', 'expstart_mjd',
+                                               'entry_date', 'mean', 'median', 'frac_masked'])
+        df_orig = df_orig.drop_duplicates(subset='filename', keep="last")  # remove any duplicate filename entries, keep the most recent
+
+        # Use the same time xlimits/xticks for all plots
+        start_mjd = 59650  # March 2022, middle of commissioning
+        end_mjd = Time.now().mjd + 0.05 * (Time.now().mjd - start_mjd)
+        time_tick_vals = np.linspace(start_mjd, end_mjd, 5)
+        time_tick_labels = [Time(m, format='mjd').isot.split('T')[0] for m in time_tick_vals]
+
+        # Make backgroud trending plots for all wide filters
+        for fltr in ['F070W', 'F090W', 'F115W', 'F150W', 'F200W', 'F277W', 'F356W', 'F444W']:
+            logging.info('Working on background trending plots for {}'.format(fltr))
+            found_limits = False
+            if int(fltr[1:4]) < 250:  # i.e. SW
+                detectors_to_run = ['NRCA2', 'NRCA4', 'NRCB3', 'NRCB1', 'NRCA1', 'NRCA3', 'NRCB4', 'NRCB2']   # in on-sky order, don't change order
+                grid = plt.GridSpec(2, 4, hspace=.4, wspace=.4, width_ratios=[1, 1, 1, 1])
+                fig = plt.figure(figsize=(40, 20))
+                fig.suptitle(fltr, fontsize=70)
+                frack_masked_thresh = 0.075
+            else:  # i.e. LW
+                detectors_to_run = ['NRCALONG', 'NRCBLONG']
+                grid = plt.GridSpec(1, 2, hspace=.2, wspace=.4, width_ratios=[1, 1])
+                fig = plt.figure(figsize=(20, 10))
+                fig.suptitle(fltr, fontsize=70, y=1.05)
+                frack_masked_thresh = 0.15
+            for i, det in enumerate(detectors_to_run):
+                logging.info('Working on {}'.format(det))
+
+                # Get relevant data for this filter/detector and remove bad datasets, e.g. crowded fields,
+                # extended objects, nebulas, short exposures.
+                df = df_orig[(df_orig['filter'] == fltr) & (df_orig['pupil'] == 'CLEAR') & (df_orig['detector'] == det) &
+                             (df_orig['effexptm'] > 300) & (df_orig['frac_masked'] < frack_masked_thresh) &
+                             (abs(1 - (df_orig['mean'] / df_orig['median'])) < 0.05)]
```

Comment:
This is for filtering images that have lots of pixels that contain signal from objects, leaving fewer pixels that are background only?

Reply:
Yeah, this omits any images that have too many pixels flagged as a source as well as shorter exposures or those where the mean/median differ (nebulas, extended objects, etc). This is to avoid bad background measurements from showing on the  background trending plots.